src/k8s/pkg/k8s/setup/serviceargs.go

```diff
@@ -2,19 +2,44 @@ package setup
 
 import (
 	"fmt"
+	"os"
+	"regexp"
+	"strings"
 
+	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/k8s/pkg/snap"
 	"github.com/canonical/k8s/pkg/utils"
 )
 
+const argumentPattern = `--?($ARGUMENT_NAME)[[:space:]]*=[[:space:]]*([^[:space:]\n]+)`
+
 // InitServiceArgs handles the setup of services arguments.
-//   - For each service, copies the default arguments files from the snap under $SNAP_DATA/args
+//   - For each service, copies the default arguments files from the snap under $SNAP_DATA/args and apply any overwrites
 //   - Note that the `k8sd` service is already configured in the snap install hook and thus not included here
-func InitServiceArgs() error {
+func InitServiceArgs(overwrites *apiv1.ExtraServiceArgs) error {
+	if overwrites == nil {
+		overwrites = &apiv1.ExtraServiceArgs{}
+	}
+
 	for _, service := range []string{"containerd", "k8s-dqlite", "kube-apiserver", "kube-controller-manager", "kube-proxy", "kube-scheduler", "kubelet"} {
```

Comment:
If the reason for a single `ExtraServiceArgs` is to do this, see https://github.com/canonical/microk8s-cluster-agent/blob/main/pkg/k8sinit/apply.go#L68-L98 for a way to do it with named struct fields as well.